game-core/src/main/java/games/strategy/triplea/ResourceLoader.java

```diff
@@ -50,12 +50,15 @@ public static ResourceLoader getGameEngineAssetLoader() {
    * Returns a resource loader that will find assets in a map directory.
    */
   public static ResourceLoader getMapResourceLoader(final String mapName) {
+    @Nullable
     File atFolder = ClientFileSystemHelper.getRootFolder();
     File resourceFolder = new File(atFolder, RESOURCE_FOLDER);
 
-    while (!resourceFolder.exists() && !resourceFolder.isDirectory()) {
+    while ((atFolder != null) && !resourceFolder.exists() && !resourceFolder.isDirectory()) {
```

Comment:
I had to convince myself my logic is right here, so think about this for yourself and correct me if I'm wrong.
I believe the `isDirectory()` check is redundant, if my understanding is correct that a file can only be a directory if it exists.
This is my pseudo-logic notation:
```
!exis && !isDir
!(exis || isDir)
!(exis || (exis && dirEntry))
!exis && !(exis && dirEntry)
!exis && (!exis || !dirEntry)
(!exis && !exis) || (!exis && !dirEntry)
!exis || !(exis || dirEntry)
!exis
```
`exis <=> exists()`
`isDir <=> isDirectory() <=> (exists() && dirEntry /* some file system internal check for the flag or something */)`
Because this doesn't really make sense I'd vote to exclusively check if it's a directory because it makes more sense.